SmartDeviceLink/SDLMenuManager.m

```diff
@@ -368,11 +368,9 @@ - (SDLAddCommand *)sdl_commandForMenuCell:(SDLMenuCell *)cell withArtwork:(BOOL)
     return command;
 }
 
-- (SDLAddSubMenu *)sdl_subMenuCommandForMenuCell:(SDLMenuCell *)cell position:(UInt16)position {
-    SDLAddSubMenu *submenu = [[SDLAddSubMenu alloc] initWithId:cell.cellId menuName:cell.title];
-    submenu.position = @(position);
-
-    return submenu;
+- (SDLAddSubMenu *)sdl_subMenuCommandForMenuCell:(SDLMenuCell *)cell withArtwork:(BOOL)shouldHaveArtwork position:(UInt16)position {
```

Comment:
If sending *two* submenus, one with artwork and one without, the manager will add an image without a name to the the `AddSubMenu` RPC. The command then fails because the image is missing a name. 

The following method should also check if `cell.icon.name` is `nil` and not just check if `shouldHaveArtwork` is true.

`SDLImage *icon = shouldHaveArtwork ? [[SDLImage alloc] initWithName:cell.icon.name] : nil;`

Reply:
Fixed